video/out/vo_dmabuf_wayland.c

```diff
@@ -709,7 +710,7 @@ static int reconfig(struct vo *vo, struct mp_image *img)
     vo->target_params = &p->target_params;
     mp_mutex_unlock(&vo->params_mutex);
 
-    wl_surface_set_buffer_transform(vo->wl->video_surface, img->params.rotate / 90);
+    wl_surface_set_buffer_transform(vo->wl->video_surface, img->params.rotate / 90 + 4 * !!img->params.vflip);
```

Comment:
This almost works. On the samples in https://github.com/ianare/exif-samples/tree/master/jpg/orientation, 2 and 4 are upside down. It unfortunately looks like the ordering of the [wayland transform enums](https://wayland.freedesktop.org/docs/html/apa.html#protocol-spec-wl_output-enum-transform) swapped these two cases so a simple `+4` won't work.

Reply:
When you say they are up-side-down, are they mirrored vertically, or are they rotated 180 degrees?

Also, according to the spec linked (which is where I referenced the +4), this should work, I wonder if the spec is wrong?

`landscape_2` results in a displaymatrix:
```
-65536    0   0
   0   65536 0
   0          0   X
```

Which has rotate set to 180 and vflip also set to `true`, which surprises me that this doesn't work.

Reply:
They're rotated 180 degrees. flipping enum 4 and 6 give the expected result

Reply:
Yeah that. I tried sway/wlroots and weston and both have the same behavior.

Reply:
Does it work now? Added a simple LUT.

Reply:
Yep it's correct now